openquake/hazardlib/tests/gsim/zhao_2016_perg_test.py

```diff
@@ -0,0 +1,322 @@
+        idx = 0 if direction == 'negative' else 1
+        azi = (strike+toward_azimuth+180) % 360
+        pointsn = npoints_towards(lon, lat, depth, azi, hdist, vdist, npoints)
+
+    sites = []
+    keys = set(site_props.keys()) - set(['vs30', 'z1pt0', 'z2pt5'])
+
+    if len(pointsn):
+        for lon, lat in reversed(pointsn[0][idx:], pointsn[1])[idx:]:
+            site = Site(Point(lon, lat, 0.0), vs30=site_props['vs30'],
+                        z1pt0=site_props['z1pt0'], z2pt5=site_props['z2pt5'])
+            for key in list(keys):
+                setattr(site, key, site_props[key])
+            sites.append(site)
+
+    for lon, lat in zip(pointsp[0], pointsp[1]):
+        site = Site(Point(lon, lat, 0.0), vs30=site_props['vs30'],
+                    z1pt0=site_props['z1pt0'], z2pt5=site_props['z2pt5'])
+        for key in list(keys):
+            setattr(site, key, site_props[key])
+        sites.append(site)
+
+    return SiteCollection(sites)
+
+def get_rupture(lon, lat, dep, msr, mag, aratio, strike, dip, rake, trt,
+                ztor=None):
+    """
+    Creates a rupture given the hypocenter position
+    """
+    hypoc = Point(lon, lat, dep)
+    srf = PlanarSurface.from_hypocenter(hypoc, msr, mag, aratio, strike, dip,
+                                        rake, ztor)
+    rup = BaseRupture(mag, rake, trt, hypoc, srf)
+    rup.hypo_depth = dep
+    return rup
+
+def get_gms_from_ctx(imt, rup, sites, gmm_perg, gmm, azimuth):
+    """
+    Get ground-motion with and without non-ergodic path effect modifications 
+    and create trellis plots to visually inspect
+    """
+    # Create context for computation of ground-motions
+    oqp = {'imtls': {k: [] for k in [str(imt)]}, 'mags': [f'{rup.mag:.2f}']}
+    
+    # Get perg version ground-motions
+    ctxm_perg = ContextMaker(gmm_perg.DEFINED_FOR_TECTONIC_REGION_TYPE,
+                             [gmm_perg], oqp)
+    
+    ctxs_perg = list(ctxm_perg.get_ctx_iter([rup], sites)) 
+    ctxs_perg = ctxs_perg[0]
+    ctxs_perg.occurrence_rate = 0.0
+    mean_perg, std_perg, tau_perg, phi_perg = ctxm_perg.get_mean_stds(
+        [ctxs_perg])
+
+    # Get non-perg version ground-motions
+    ctxm = ContextMaker(gmm.DEFINED_FOR_TECTONIC_REGION_TYPE, [gmm], oqp)
+    ctxs = list(ctxm.get_ctx_iter([rup], sites)) 
+    ctxs = ctxs[0]
+    
+    ctxs.occurrence_rate = 0.0
+    mean, std, tau, phi = ctxm.get_mean_stds([ctxs])
+    
+    # Plot perg vs non-perg predicted gm vs rjb
+    dist_x = ctxs.rrup
+    mean_perg = mean_perg[0][0]
+    mean = mean[0][0]
+    pyplot.plot(dist_x, np.exp(mean_perg), 'r', label = 'PErg')
+    pyplot.plot(dist_x, np.exp(mean), 'b', label = 'Non-PErg')
+    pyplot.semilogy()
+    pyplot.ylabel('%s (g)' %(imt))
+    pyplot.xlabel('Joyner-Boore distance (km)')
+    pyplot.title('Test scenario for imt = %s and site azimuth = %s$^o$' %(
+        imt, azimuth))
+    pyplot.legend()
+    pyplot.show()
+    
+    return mean_perg, mean
+    
```

Comment:
There are ~200 lines of test code which are not necessary since the engine already has the functionality of building contexts from a rupture, see the method cmaker.from_planar used here: https://github.com/gem/oq-engine/blob/master/openquake/hazardlib/tests/contexts_test.py#L546C5-L546C5